src/utils.cairo

```diff
@@ -1,6 +1,50 @@
-use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
 use core::num::traits::{Zero, One, BitSize};
-use core::starknet::secp256_trait::Secp256PointTrait;
+use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+
+#[derive(Copy, Drop, Debug)]
+// pub is required here as we define Hash in utils and we need to import Hash in merkle_tree.cairo
+// for arguments to merkle_root function
+pub struct Hash {
+    value: [u32; 8]
+}
+
+#[generate_trait]
+pub impl HashImpl of HashTrait {
+    #[inline(always)]
+    fn new() -> Hash {
+        Hash { value: [0; 8] }
+    }
+}
+
+pub impl U256IntoHash of Into<u256, Hash> {
+    fn into(self: u256) -> Hash {
+        let mut result: Array<u32> = array![];
+        let mut value = self;
+
+        let mut i = 0;
+        loop {
```

Comment:
Shr high and low word of u256 separately, `shr` on `u256` costs more than shr on `u128`.

Reply:
done!